# Question 1: 

Given some sample data, write a program to answer the following: [click here to access the required data set](https://docs.google.com/spreadsheets/d/16i38oonuX1y1g7C_UAmiK9GkY7cS-64DfiDMNiR41LM/edit#gid=0)

<br>
On Shopify, we have exactly 100 sneaker shops, and each of these shops sells only one model of shoe. We want to do some analysis of the average order value (AOV). When we look at orders data over a 30 day window, we naively calculate an AOV of $3145.13. Given that we know these shops are selling sneakers, a relatively affordable item, something seems wrong with our analysis. 


In [4]:
import pandas as pd

df = pd.read_csv('challenge.csv')
df.head(25)

order_id  shop_id  user_id  order_amount  total_items payment_method  \
0          1       53      746           224            2           cash   
1          2       92      925            90            1           cash   
2          3       44      861           144            1           cash   
3          4       18      935           156            1    credit_card   
4          5       18      883           156            1    credit_card   
5          6       58      882           138            1    credit_card   
6          7       87      915           149            1           cash   
7          8       22      761           292            2           cash   
8          9       64      914           266            2          debit   
9         10       52      788           146            1    credit_card   
10        11       66      848           322            2    credit_card   
11        12       40      983           322            2          debit   
12        13       54      799           266            2    credit_card   
13        14      100      709           111            1           cash   
14        15       87      849           447            3    credit_card   
15        16       42      607        704000         2000    credit_card   
16        17       17      731           176            1           cash   
17        18       28      752           164            1    credit_card   
18        19       83      761           258            2           cash   
19        20       63      898           408            3    credit_card   
20        21       66      987           322            2           cash   
21        22       97      789           486            3    credit_card   
22        23       88      985           704            4    credit_card   
23        24       75      964           256            2    credit_card   
24        25       73      917           495            3           cash   

             created_at  
0   2017-03-13 12:36:56  
1   2017-03-03 17:38:52  
2    2017-03-14 4:23:56  
3   2017-03-26 12:43:37  
4    2017-03-01 4:35:11  
5   2017-03-14 15:25:01  
6   2017-03-01 21:37:57  
7    2017-03-08 2:05:38  
8   2017-03-17 20:56:50  
9   2017-03-30 21:08:26  
10  2017-03-26 23:36:40  
11  2017-03-12 17:58:30  
12  2017-03-16 14:15:34  
13   2017-03-22 2:39:49  
14  2017-03-10 11:23:18  
15   2017-03-07 4:00:00  
16   2017-03-21 4:23:38  
17  2017-03-21 12:09:07  
18  2017-03-17 13:18:47  
19  2017-03-29 15:11:52  
20  2017-03-30 20:11:59  
21  2017-03-04 15:44:00  
22   2017-03-22 1:19:41  
23   2017-03-12 3:07:42  
24  2017-03-03 13:01:03

In [5]:
df['order_amount'].describe()

count      5000.000000
mean       3145.128000
std       41282.539349
min          90.000000
25%         163.000000
50%         284.000000
75%         390.000000
max      704000.000000
Name: order_amount, dtype: float64

A. Think about what could be going wrong with our calculation. Think about a better way to evaluate this data. 

As the mean and standard deviation for this data are so huge, we can conclude that data are more spread out. As a result, the mean is a poor estimate of AOV.
Using the median as the AOV is a better way to examine the data as it is a central number and is not affected by the outliers in contrast with the mean value.

B. What metric would you report for this dataset?

For this dataset, the median would be a better metric to report.

C. What is its value?

In [6]:
df['order_amount'].median()

284.0

The better value to report as AOV is the median of order_amount which its value is $284.

# Question 2

For this question you’ll need to use SQL. [Follow this link](https://www.w3schools.com/SQL/TRYSQL.ASP?FILENAME=TRYSQL_SELECT_ALL) to access the data set required for the challenge. Please use queries to answer the following questions. Paste your queries along with your final numerical answers below.

A. How many orders were shipped by Speedy Express in total?

SELECT COUNT(*) FROM Orders

JOIN Shippers ON Shippers.ShipperID = Orders.ShipperID

WHERE Shippers.ShipperName = 'Speedy Express';

After joining two tables of orders and shippers, we can reach the number of orders Speedy Express has shipped, which is 54.

B. What is the last name of the employee with the most orders?

SELECT Employees.LastName, Count() AS TotalOrders FROM Orders

JOIN Employees ON Orders.EmployeeID = Employees.EmployeeID

GROUP BY Employees.LastName ORDER BY TotalOrders DESC

Limit 1;


<img src='https://drive.google.com/uc?id=1TCpMZdZgYrv5tz-L6tgn_aguASn4YEYH'>


By joining two tables of orders and employees, and then grouping by the employee last name, we can figure out that Peacock is the employee who has the most orders (40 orders).


C. What product was ordered the most by customers in Germany?

JOIN OrderDetails ON Orders.OrderID = OrderDetails.OrderID

JOIN Customers ON Orders.CustomerID = Customers.CustomerID

JOIN Products ON OrderDetails.ProductID = Products.ProductID

WHERE Country = 'Germany'

GROUP BY Products.ProductName ORDER BY TotalOrdered DESC

Limit 1;

<img src='https://drive.google.com/uc?id=1d27mIDOv8OKFUXisxVQ3I8Eg94y_K1YH'>


By joining tables of order, orderdetails, customer, and product and sum the order quantities for products ordered by Germany, we can figure out that Boston Crab Meat is  the most popular item (with 160 orders).